src/Symfony/Component/Security/Http/EventListener/CsrfTokenClearingLogoutListener.php

```diff
@@ -32,7 +32,12 @@ public function __construct(ClearableTokenStorageInterface $csrfTokenStorage)
 
     public function onLogout(LogoutEvent $event): void
     {
-        if ($this->csrfTokenStorage instanceof SessionTokenStorage && !$event->getRequest()->hasPreviousSession()) {
+        $request = $event->getRequest();
+
+        if (
+            $this->csrfTokenStorage instanceof SessionTokenStorage
+            && ($request->attributes->getBoolean('_stateless') || !$request->hasPreviousSession())
```

Comment:
But why do we have a previous request in the first place in your situation?
If there is a previous session, we should clean the session even of this contradicts the stateless flag, or we might create a security issue, or?

Reply:
Our API and website are sharing the same domain.
API is accessible from `domain.com/api/`
The user can be authenticated on the website using session and also using JWT on the API on the same device.

When the user already has a session on the website and try to logout from the API, it results in this log message.
(On the API we use the logout endpoint to delete refresh tokens of the user in our db)
Because the logout listener is only checking if the request ha a previous session when all other listeners also check if the request is stateless.

Reply:
BTW, this is also how works others logout listeners.
If the user logout on a stateless request (and also has a session on the same device), his session is not cleared.
So we should also clear the CSRF token too on statleless requests.

Reply:
is the API using the same firewall than the website ?

Reply:
No, we had a separated stateless firewall for `^/api` routes.
I was not able to find how to reproduce that issue myself by using our mobile app, but I was suspecting the mobile app to call some website routes in some rare scenarios.

Reply:
then I think the right fix would be that the CSRF clearing logout listener should be registered only for stateful **firewalls** (note that I'm not talking about the flag on the request here, but on the firewall config).